### PR TITLE
errno: add ETXTBSY

### PR DESCRIPTION
**Pull request by indutny**
Add `ETXTBSY` which could be returned by file system reads on some
unixes and emulate it with -148 (`ERROR_PATH_BUSY`) on windows.

/cc @saghul

### CODE REVIEW

File: include/uv-errno.h [hunk 1]
```diff
@@ -364,6 +364,12 @@
 # define UV__ETIMEDOUT (-4039)
 #endif
 
+#if defined(ETXTBSY) && !defined(_WIN32)
+# define UV__ETXTBSY (-ETXTBSY)
+#else
+# define UV__ETXTBSY (-148)
```
Comment (saghul) [inline include/uv-errno.h]: Why -148? On Windows we remap all errors ourselfs, so shouldn't it be -4038?

Comment (indutny) [inline include/uv-errno.h]: Oh, didn't know about that. Perhaps, you are right.

Comment (saghul) [inline include/uv-errno.h]: You can add the windows mapping here: https://github.com/joyent/libuv/blob/master/src/win/error.c#L67

**Comment by indutny** [comment]
@saghul force pushed.

File: include/uv-errno.h (hunk 1 above)
Comment (indutny) [inline include/uv-errno.h]: I guess it is fine to leave it unmapped.

**Comment by saghul** [comment]
LGTM.

**Comment by indutny** [comment]
Oh, actually it should land in v0.10 first.

**Comment by saghul** [comment]
Well, I can backport it with the other 2, no worries.

**Comment by indutny** [comment]
Ok, thank you, landed in aaaefe3

**Comment by saghul** [comment]
Landed also in v0.10

**Pull request closed without merging**